### ISSUE TITLE
`@Query` on a void method

### ISSUE THREAD

**Issue by andymc12** [eclipse/microprofile-graphql#85]
In last week's meeting we discussed what currently happens and what _should_ happen when a user puts an `@Query` annotation on a void method.

The current behavior seems to be that it would generate a schema with Boolean return type and then return Boolean.TRUE (assuming no errors).

The desired behavior is that this is a deployment error and should prevent the application from starting. 

To resolve this issue, we should document the behavior and if possible add a TCK test for it.

**Comment by phillip-kruger** [comment]
@andymc12 - Should we not just print a warning ? Example, when you put a `NonNull`  and a `DefaultValue` we ignore the `NonNull` and print a warning, but we do not fail ?

**Comment by andymc12** [comment]
I thought we decided that it was a deployment error in our meeting on October 11 meeting - I think the part of the reasoning was that there was no good use case for having a void query method - and that we could undo this restriction in a later release without causing breaking changes, but not the reverse (i.e. it would be a breaking change if we started enforcing this restriction in a later release after previously allowing it).  

The `@NonNull` plus `@DefaultValue` case is more of a warning - or even just an unnecessary annotation - if there is a default value, then the value would never be null when executing the method.  So I think this case is ok to allow.
